HCDevice.py

```diff
@@ -92,19 +93,62 @@ def parse_values(self, values):
 
 		return result
 
+	# Test the feature of an appliance agains a data object
+	def test_feature(self, data):
+		if 'uid' not in data:
+			raise Exception("{self.name}. Unable to configure appliance. UID is required.")
+
+		if isinstance(data['uid'], int) == False:
+			raise Exception("{self.name}. Unable to configure appliance. UID must be an integer.")
+
+		if 'value' not in data:
+			raise Exception("{self.name}. Unable to configure appliance. Value is required.")
+
+		# Check if the uid is present for this appliance
+		uid = str(data['uid'])
+		if uid not in self.features:
+			raise Exception(f"{self.name}. Unable to configure appliance. UID {uid} is not valid.")
+
+		feature = self.features[uid]
+
+		# check the access level of the feature
+		print(now(), self.name, f"Processing feature {feature['name']} with uid {uid}")
+		if 'access' not in feature:
+			raise Exception(f"{self.name}. Unable to configure appliance. Feature {feature['name']} with uid {uid} does not have access.")
+
+		access = feature['access'].lower()
+		if access != 'readwrite' and access != 'writeonly':
+			raise Exception(f"{self.name}. Unable to configure appliance. Feature {feature['name']} with uid {uid} has got access {feature['access']}.")
+
+		# check if selected list with values is allowed
+		if 'values' in feature:
+			if isinstance(data['value'], int) == False:
+				raise Exception(f"Unable to configure appliance. The value {data['value']} must be an integer. Allowed values are {feature['values']}.")
+			value = str(data['value']) # values are strings in the feature list, but always seem to be an integer. An integer must be provided
+			if value not in feature['values']:
+				raise Exception(f"{self.name}. Unable to configure appliance. Value {data['value']} is not a valid value. Allowed values are {feature['values']}.")
+
+		if 'min' in feature:
+			min = int(feature['min'])
+			max = int(feature['max'])
+			if isinstance(data['value'], int) == False or data['value'] < min or data['value'] > max:
```

Comment:
comparison to False should be 'if cond is False:' or 'if not cond:'
https://www.flake8rules.com/rules/E712.html

Or is there a reason for this implementation?